modules/apps/frontend-icons/frontend-icons-web/src/main/resources/META-INF/resources/js/site_settings/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import ClayPanel from '@clayui/panel';
+import React, {useMemo, useState} from 'react';
+
+import AddIconPackModal from './AddIconPackModal';
+
+export default function SiteIconPackConfiguration({
+	allIconResourcePacks: icons,
+	portletNamespace,
+	saveSiteIconPacksURL,
+	siteIconResourcePacks,
+}) {
+	const [selectedPacks, setSelectedPacks] = useState(siteIconResourcePacks);
+	const [searchQuery, setSearchQuery] = useState('');
+	const [showAddModal, setShowAddModal] = useState(false);
+
+	const iconPackNames = Object.keys(icons);
+
+	const filteredIcons = useMemo(() => {
+		return iconPackNames.reduce((acc, packName) => {
+			return {
+				...acc,
+				[packName]: {
+					...icons[packName],
+					icons: icons[packName].icons
+						.filter((icon) => Object.keys(icon).length)
+						.filter((icon) =>
+							icon.name
+								.toLowerCase()
+								.includes(searchQuery.toLocaleLowerCase())
+						),
+				},
+			};
+		}, {});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [iconPackNames, icons, searchQuery]);
+
+	const referenceTime = useMemo(
+		() => new Date().getTime(),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[icons]
+	);
+
+	return (
+		<>
+			<ClayLayout.ContentRow containerElement="p">
+				{Liferay.Language.get('icons-spritemap-path-at')}
+			</ClayLayout.ContentRow>
+
+			<ClayLayout.ContentRow className="mb-2" containerElement="code">
+				{window.location.host +
+					`/o/icons/site/${Liferay.ThemeDisplay.getSiteGroupId()}.svg`}
+			</ClayLayout.ContentRow>
+
+			<label className="form-control-label">
+				<span className="form-control-label-text">
+					{Liferay.Language.get('search-available-icons')}
+				</span>
+
+				<ClayInput
+					onChange={(event) => setSearchQuery(event.target.value)}
+					placeholder={Liferay.Language.get('search-available-icons')}
+					type="text"
+					value={searchQuery}
+				/>
+			</label>
+
+			<ClayPanel.Group className="mt-4">
+				<span className="form-control-label-text">
+					{Liferay.Language.get('available-icon-packs')}
+				</span>
+
+				{iconPackNames
+					.filter((name) => selectedPacks.includes(name))
```

Comment:
Hey @bryceosterhaus, how many items are expected to be at selectedPacks and inside iconPackNames? filter -> includes is quadratic. Maybe we can use a dict/Map on `initialSelectedPacks`

Reply:
@diegonvs I wouldn't really expect more than 10, even that would be large though, so I think this should be fine.